gemrb/core/Scriptable/Actor.cpp

```diff
@@ -529,7 +529,7 @@ Actor::stat_t Actor::GetSafeStat(unsigned int StatIndex) const
 	if (StatIndex >= MAX_STATS) {
 		return 0xdadadada;
 	}
-	if (PrevStats) return PrevStats[StatIndex];
+	//if (PrevStats) return PrevStats[StatIndex];
```

Comment:
definitely bad

Reply:
Hmm, I can reproduce this just walking on some random traps. The core issue is likely somewhere else though, this used to work fine. I can confirm it did in 0.9.1.

Reply:
Ok, found the issue — in the effect itself. It works fine unless you get poisoned the same way twice, which is not unlikely when getting jumped by several spiders.